it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersPartnershipCompanyRegistrationNumberControllerISpec.scala

```diff
@@ -1,119 +1,165 @@
 package uk.gov.hmrc.fhregistrationfrontend.controllers
 
-import models.NormalMode
+import models.{CheckMode, Mode, NormalMode, UserAnswers}
 import org.jsoup.Jsoup
+import org.scalatest.TryValues.convertTryToSuccessOrFailure
 import play.api.libs.ws.DefaultWSCookie
 import play.mvc.Http.HeaderNames
+import uk.gov.hmrc.fhregistrationfrontend.forms.models.CompanyRegistrationNumber
+import uk.gov.hmrc.fhregistrationfrontend.pages.businessPartners.CompanyRegistrationNumberPage
 import uk.gov.hmrc.fhregistrationfrontend.testsupport.{Specifications, TestConfiguration}
 
 class BusinessPartnersPartnershipCompanyRegistrationNumberControllerISpec
-  extends Specifications with TestConfiguration {
-
-  val route: String = routes.BusinessPartnersPartnershipCompanyRegistrationNumberController.load().url.drop(6)
-  val partnershipVatNumUrl: String = routes.BusinessPartnersPartnershipVatNumberController.load(1, NormalMode).url
-
-  s"GET $route" should {
-    "render the partnership-company-registration-number page" in {
-      given
-        .commonPrecondition
-
-      val result = buildRequest(route)
-        .addCookies(
-          DefaultWSCookie("mdtp", authAndSessionCookie),
-          DefaultWSCookie("businessType", "limited-liability-partnership")
-        )
-        .get()
-
-      whenReady(result) { res =>
-        res.status mustBe 200
-        val page = Jsoup.parse(res.body)
-        page.title() must include("What is the partnership’s company registration number?")
-        page.getElementsByTag("h1").text() must include("What is Test Partnership’s company registration number?")
-      }
-    }
-  }
+    extends Specifications with TestConfiguration {
+
+  def route(mode: Mode): String =
+    routes.BusinessPartnersPartnershipCompanyRegistrationNumberController.load(1, mode).url.drop(6)
+  def partnershipVatNumUrl(mode: Mode): String = routes.BusinessPartnersPartnershipVatNumberController.load(1, mode).url
+  val userAnswersWithPageData: UserAnswers = emptyUserAnswers
+    .set[CompanyRegistrationNumber](CompanyRegistrationNumberPage(1), CompanyRegistrationNumber("01234567", None))
+    .success
+    .value
+
+  List(NormalMode, CheckMode).foreach { mode =>
+    s"GET ${route(mode)}" when {
+      "render the partnership-company-registration-number page with " when {
+        "there are useranswers but no form data" in {
```

Comment:
Test sentences could be clearer 

'with' at the end of the second sentence is not needed 
'render the partnership-company-registration-number page with' to 'render the partnership-company-registration-number page'.

Also 'useranswers' to 'user answers'.